api/types/types_paths.go

```diff
@@ -121,43 +191,88 @@ func (k fileKey) key() string {
 		return "log"
 	case Run:
 		return "run"
+	case TLS:
+		return "tls"
 	case LSX:
 		return "lsx"
 	}
 	return ""
 }
 
+func isLibstorageHomeSet() bool {
+	return libstorageHome != "/"
+}
+
 func (k fileKey) defaultVal() string {
+	if v, ok := k.isEnvVarSet(); ok {
+		return v
+	}
 	switch k {
 	case Home:
 		return libstorageHome
 	case Etc:
+		if isLibstorageHomeSet() {
+			return "/etc"
+		}
 		return "/etc/libstorage"
 	case Lib:
+		if isLibstorageHomeSet() {
+			return "/var/lib"
+		}
 		return "/var/lib/libstorage"
 	case Log:
+		if isLibstorageHomeSet() {
+			return "/var/log"
+		}
 		return "/var/log/libstorage"
 	case Run:
+		if isLibstorageHomeSet() {
+			return "/var/run"
+		}
 		return "/var/run/libstorage"
+	case TLS:
+		return "tls"
```

Comment:
Why is TLS different from the others, and not ever dependent on LIBSTORAGE_HOME ?